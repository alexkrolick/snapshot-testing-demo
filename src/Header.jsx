import React from 'react';

const SignedIn = () => 'Welcome!'
const SignedOut = () => 'Please sign in'

const Header = ({signedIn = false}) => {
  return (
    <div>
      {
        signedIn ?  <SignedIn /> :< SignedOut />
      }
    </div>
  );
};

export default Header;