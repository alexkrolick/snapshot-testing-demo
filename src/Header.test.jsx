import React from 'react'
import Header from './Header'
import { create as render } from "react-test-renderer";
import { shallow } from 'enzyme'

it("should render signed out by default", () => {
  const snap = shallow(<Header />);
  expect(snap).toMatchSnapshot();
});

it("should render signed in", () => {
  const snap = shallow(<Header signedIn />);
  expect(snap).toMatchSnapshot();
});