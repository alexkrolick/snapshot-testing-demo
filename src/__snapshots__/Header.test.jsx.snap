// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render signed in 1`] = `
<div>
  <SignedIn />
</div>
`;

exports[`should render signed out by default 1`] = `
<div>
  <SignedOut />
</div>
`;
